proto/cosmos/distribution/query.proto

```diff
@@ -5,37 +5,56 @@ import "cosmos/query/pagination.proto";
 import "gogoproto/gogo.proto";
 import "cosmos/cosmos.proto";
 import "cosmos/distribution/distribution.proto";
+import "google/api/annotations.proto";
 
 option go_package = "github.com/cosmos/cosmos-sdk/x/distribution/types";
 
 // Query defines the gRPC querier service for distribution module
 service Query {
   // Params queries params of distribution module
-  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {}
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/cosmos/distribution/params";
+  }
 
   // ValidatorOutstandingRewards queries rewards of a validator address
-  rpc ValidatorOutstandingRewards(QueryValidatorOutstandingRewardsRequest) returns (QueryValidatorOutstandingRewardsResponse) {}
+  rpc ValidatorOutstandingRewards(QueryValidatorOutstandingRewardsRequest) returns (QueryValidatorOutstandingRewardsResponse) {
+    option (google.api.http).get = "/cosmos/distribution/{validator_address}/outstanding-rewards";
+  }
 
   // ValidatorCommission queries accumulated commission for a validator
-  rpc ValidatorCommission (QueryValidatorCommissionRequest) returns (QueryValidatorCommissionResponse) {}
+  rpc ValidatorCommission (QueryValidatorCommissionRequest) returns (QueryValidatorCommissionResponse) {
+    option (google.api.http).get = "/cosmos/distribution/{validator_address}/validator-commission";
+  }
 
   // ValidatorSlashes queries slash events of a validator
-  rpc ValidatorSlashes (QueryValidatorSlashesRequest) returns (QueryValidatorSlashesResponse) {}
+  rpc ValidatorSlashes (QueryValidatorSlashesRequest) returns (QueryValidatorSlashesResponse) {
+    option (google.api.http).get = "/cosmos/distribution/{validator_address}/validator-slashes/{starting_height}/{ending_height}/{pagination.key}/{pagination.offset}/{pagination.limit}/{pagination.count_total}";
```

Comment:
I would propose to pass all optional config as query params `?pageOffset=...&...`

Reply:
I think adding query params are not necessary in annotations. I removed them from the routing, they can be sent when the api called.